custom/icds_reports/utils/aggregation_helpers/distributed/bihar_api_child_vaccine.py

```diff
@@ -0,0 +1,114 @@
+
+    def aggregate(self, cursor):
+        drop_query = self.drop_table_query()
+        create_query = self.create_table_query()
+        agg_query = self.aggregation_query()
+        index_queries = self.indexes()
+        add_partition_query = self.add_partition_table__query()
+
+        cursor.execute(drop_query)
+        cursor.execute(create_query)
+        cursor.execute(agg_query)
+        for query in index_queries:
+            cursor.execute(query)
+
+        cursor.execute(add_partition_query)
+
+    def drop_table_query(self):
+        return f"""
+                DROP TABLE IF EXISTS "{self.monthly_tablename}"
+            """
+
+    def create_table_query(self):
+        return f"""
+            CREATE TABLE "{self.monthly_tablename}" (LIKE {self.tablename});
+            SELECT create_distributed_table('{self.monthly_tablename}', 'supervisor_id');
+        """
+
+    @property
+    def monthly_tablename(self):
+        return f"{self.tablename}_{month_formatter(self.month)}"
+
+    @property
+    def bihar_state_id(self):
+        return SQLLocation.objects.get(name='Bihar', location_type__name='state').location_id
+
+    def aggregation_query(self):
+        month_start_string = month_formatter(self.month)
+        month_end_string = month_formatter(self.month + relativedelta(months=1, seconds=-1))
+        person_case_ucr = get_table_name(self.domain, 'static-person_cases_v3')
+        ccs_case_ucr = get_table_name(self.domain, 'static-ccs_record_cases'),
+        child_health_monthly = get_child_health_tablename(self.month)
+
+        columns = (
+            ('month', f"'{month_start_string}'"),
+            ('person_id', 'person_list.doc_id'),
+            ('supervisor_id', 'person_list.supervisor_id'),
+            ('time_birth', 'person_list.time_birth'),
+            ('child_alive', 'person_list.child_alive'),
+            ('father_name', 'person_list.father_name'),
+            ('mother_name', 'person_list.mother_name'),
+            ("case_id", "child_health.case_id"),
+            ('dob', 'person_list.dob'),
+            ('household_case_id', 'person_list.household_case_id'),
+            ('delivery_nature', 'ccs_cases.delivery_nature'),
+            ('term_days', 'CASE WHEN (person_list.dob-ccs_cases.lmp >= 0) THEN 1 ELSE NULL END')
+        )
+        column_names = ", ".join([col[0] for col in columns])
+        calculations = ", ".join([col[1] for col in columns])
+
+        return f"""
+                INSERT INTO "{self.monthly_tablename}" (
+                    {column_names}
+                )
+                (
+                SELECT
+                {calculations}
+                from "{child_health_monthly}" child_health
+                LEFT JOIN "{person_case_ucr}" person_list ON (
+                    person_list.doc_id = child_health.mother_id
+                    AND person_list.supervisor_id = child_health.supervisor_id
+                )
+                LEFT JOIN "{ccs_case_ucr}" ccs_cases ON (
```

Comment:
This join will not work. A given person can have multiple CCS record cases for multiple different children. As such, this will create lots of duplicate rows. If a mother has 4 ccs record cases and 4 children (there will likely be the same number since this does not exclude closed ccs records, although its possible that there will be fewer ccs record cases if some of the children were registered after turning 3), then 16 rows will be created for those children rather than 4, one for each ccs_record, child_health pair.

This kind of join (between ccs and child health) is generally an operation we don't support. Is there another way to get the data you need out of this.

Reply:
@calellowitz  did not actually follow how that could happen because doc_id is unique in person case ucr. But lets offline about this

Reply:
I see what you are saying